### DISCUSSION TITLE
How to get connected wallet address and meta info?

### DISCUSSION THREAD

**Discussion by itsnyx** [WalletConnect/walletconnect-monorepo#2357] [Developer Support] [answered]
Hello, how can i get connected wallet address?

and also can i get the logo and title of the connected wallet ?

**Comment by xzilja** [comment]
How are you using web3modal? (wagmi / ethereum-provider / sign-client)?

**Reply by itsnyx** [reply]
web3modal-sdk for react native version 1.0.0-alpha.3.

https://github.com/WalletConnect/web3modal-react-native

i want to get connected wallet address and if possible connected wallet logo and name.
this library has a hook that will give provider object but I don't know how to use it properly .

**Reply by ignaciosantise** [reply, marked as answer]
@itsnyx since`@web3modal/react-native 1.0.0-alpha.5` you can access the wallet address using `useWeb3Modal` hook:

`const { address } = useWeb3Modal();`

and to get the wallet logo and name you can access the `peer metadata` from the `provider`:

```
  const { provider } = useWeb3Modal();

 const walletName = provider?.session?.peer?.metadata?.name;
 const walletIcon = provider?.session?.peer?.metadata?.icons[0];
```

Make the corresponding checks before accessing the peer metadata, as it could be `null/undefined`

**Reply by itsnyx** [reply]
thanks for answer 
and **big thanks** to your hard-work on this project
